### PR TITLE
Implement decodeFormData function and associated tests

### PR DESCRIPTION
**Pull request by fabian-hiller**
## Summary by cubic
Adds `decodeFormData(schema, formData)` to rebuild typed, nested values from `FormData` using the Valibot schema, restoring numbers, booleans, dates, arrays, and guarding against prototype pollution and array property abuse. Exported from `packages/core/src/form/decodeFormData` and re-exported via `packages/core/src/form`.

- New Features
  - Typed decoding by schema: numbers, booleans (unchecked default to false), dates (incl. `datetime-local`), and bigints; files pass through; empty unselected files are skipped.
  - Build nested objects/arrays from JSON path keys like `["todos",0,"label"]`; supports repeated-key arrays, tuples, unions/intersects/variant, wrapper and lazy schemas, and pipelines.
  - Fill defaults only within present containers: absent arrays become `[]`; booleans default to `false`.
  - Safety and limits: ignore invalid JSON keys/segments and dangerous keys (`__proto__`, `prototype`, `constructor`); ignore string properties on arrays (e.g. `"length"`, `"push"`, and numeric indices sent as strings); throw when a numeric array index would exceed 5000 items.
  - Robustness: if a parent path already holds a scalar, later nested entries for that path are ignored to avoid crashes.
  - Comprehensive tests cover decoding behavior, containers, combinators, defaults, limits, edge cases, and integration.

<sup>Written for commit 86b63c378753b0779aeba8d815bea049327f55ce. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/open-circle/formisch/pull/125?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/core/src/form/decodeFormData/decodeFormData.ts [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+    case 'optional':
+    case 'undefinedable':
+    case 'non_nullable':
+    case 'non_nullish':
+    case 'non_optional':
+      return unwrapSchema(schema.wrapped!);
+    case 'lazy':
+      return unwrapSchema(schema.getter!(undefined));
+    default:
+      return schema;
```
[the remark this answers is not in the document]
Comment (fabian-hiller) [reply packages/core/src/form/decodeFormData/decodeFormData.ts:63]: I am ok ignoring this edge case for now. What do you think?
